packages/compass-connections/src/components/legacy-connections.tsx

```diff
@@ -76,6 +86,23 @@ const formCardLightThemeStyles = css({
   background: palette.white,
 });
 
+// Single connection form is a bit of a special case where form is always on
+// screen so even when user is not explicitly editing any connections, we need a
+// connection info object to be passed around. For that purposes this hook will
+// either return an editing connection info or will create a new one as a
+// fallback if nothing is actively being edited
+function useActiveConnectionInfo(
```

Comment:
It might be something that we maybe want to just make part of the connection-form logic, it's kinda convenient to have a default there and then we don't need to manually create new connection in two different places